### PR TITLE
Add ec2_eni module (aws network interfaces)

### PR DESCRIPTION
**Pull request by wimnat**
Adds eni support for AWS.  Can be used to create, destroy, attach, detach and list ENIs.

Tested in my own environment and seems good but would appreciate some more testing by anyone else.

### CODE REVIEW

**Comment by evanccnyc** [comment]
@wimnat You should add the region option so that can be set there.

I'd love to have this checked in. Ill see who I can push to get it done.

**Comment by wimnat** [comment]
@evanccnyc specifying region does work but it's not documented explicitly as i assume "extends_documentation_fragment: aws" will include it.

**Comment by evanccnyc** [comment]
@wimnat Sorry about that I should have read more closely.

**Comment by evanccnyc** [comment]
@bcoca anything else that needs to be done for this to get merged?

File: cloud/amazon/ec2_eni.py [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+# You should have received a copy of the GNU General Public License
+# along with this library.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: ec2_eni
+short_description: Create and optionally attach an Elastic Network Interface (ENI) to an instance
+description:
+    - Create and optionally attach an Elastic Network Interface (ENI) to an instance. If an ENI ID is provided, an attempt is made to update the existing ENI. By passing 'None' as the instance_id, an ENI can be detached from an instance.
+version_added: "1.9"
```
Comment (bcoca) [inline cloud/amazon/ec2_eni.py]: 2.0

File: cloud/amazon/ec2_eni.py [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+author: Rob White, wimnat [at] gmail.com, @wimnat
+options:
+  instance_id:
+    description:
+      - Instance ID that you wish to attach ENI to 
+    required: false
+    default: null 
+    aliases: []
```
Comment (bcoca) [inline cloud/amazon/ec2_eni.py]: no need for empty aliases

File: cloud/amazon/ec2_eni.py [hunk 3]
```diff
@@ -0,0 +1,406 @@
...
+    description:
+      - Private IP address.
+    required: false
+    default: null
+    aliases: []
+  subnet_id:
+    description:
+      - Subnet in which to create the ENI
+    required: true
+    default: null
```
Comment (bcoca) [inline cloud/amazon/ec2_eni.py]: if required, default does not make sense

File: cloud/amazon/ec2_eni.py [hunk 4]
```diff
@@ -0,0 +1,406 @@
...
+  delete_on_termination:
+    description:
+      - Delete the interface when the instance it is attached to is terminated. You can only specify this flag when the interface is being modified, not on creation.
+    required: false
+  source_dest_check:
+    description:
+      - By default, interfaces perform source/destination checks. NAT instances however need this check to be disabled. You can only specify this flag when the interface is being modified, not on creation.
+    required: false  
+extends_documentation_fragment: aws
+requirements: [ "boto" ]
```
Comment (bcoca) [inline cloud/amazon/ec2_eni.py]: duplicates requirements already in doc fragments

File: cloud/amazon/ec2_eni.py (hunk 3 above)
Comment (wimnat) [inline cloud/amazon/ec2_eni.py]: @bcoca subnet_id is required for state: present but not for state: absent.  What's the best way to document?

Comment (bcoca) [inline cloud/amazon/ec2_eni.py]: in the description

**Comment by evanccnyc** [comment]
Any idea when this can get checked in?

File: cloud/amazon/ec2_eni.py [hunk 5]
```diff
@@ -0,0 +1,396 @@
...
+    description:
+      - Optional description of the ENI.
+    required: false
+    default: null
+  security_groups:
+    description:
+      - List of security groups associated with the interface. Only used when state=present.
+    required: false
+    default: null
+  state:
```
Comment (bcoca) [inline cloud/amazon/ec2_eni.py:54]: you should document possible states here with a choices: [ ...] entry, also the 'list' state is indicative of code belonging in a facts module.

**Comment by wimnat** [comment]
The list state has been moved out to a separate facts module and I've added in the choices for 'state'.  I also squashed the commits.

**Comment by gregdek** [comment]
@bcoca can you re-review?

**Pull request merged**